fix(fastify-template): fix for subscriptions in fastify template

RinkiyaKeDad opened the pull request:
<!--The content below is a suggested layout for writing your pull request description. You may extend or remove parts you deem relevant or irrelevant to your changes.-->

<!--Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to [link to that issue](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).-->
Fix for #2174 

## Types of changes

What types of changes does your code introduce to Graphback?

<!--Put an `x` in the boxes that apply-->

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please specify)

## Checklist

<!--Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.-->

- [x] I have read the [CONTRIBUTING](https://github.com/aerogear/graphback/blob/master/CONTRIBUTING.md) document.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)


## Further comments

As [discussed](https://github.com/aerogear/graphback/issues/2174#issuecomment-752543660) I've added resolvers in the plugin options of mercurius which should fix the issue.
<!--If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...-->


## Verification

I tried verifying this by enabling the fastify template and then using the updated cli (by running `node dist/index.js` in the create-graphback folder) but I received this error due to which the template didn't load and I wasn't able to test it. Please let me know if there are some other steps I can take to test if the subscriptions are working or not. Thanks.
![sendNew](https://user-images.githubusercontent.com/56963264/103439052-a04df600-4c5f-11eb-93e0-191b69e9ee6f.png)
 

<!--If there are any steps you took to verify that this works, please specify them here.-->

machi1990 commented on templates/ts-fastify-mongodb-backend/src/index.ts:
```diff
@@ -45,6 +45,7 @@ async function start() {
 
   app.register((mercurius as any), {
     schema,
+    resolvers,
```
I think `resolvers` should include `noteResolvers` as in https://github.com/aerogear/graphback/blob/12d588718dacaecb5cc1d3ed8005ac4a949b82a4/templates/ts-fastify-mongodb-backend/src/index.ts#L40

machi1990 reviewed (changes requested):
Thanks for the PR. I've not verified it as I am away from laptop but it LGTM barring the small requested change inline. 

I'll let @craicoverflow @wtrocki do the verification.

wtrocki commented:
Let's start this year with this small PR. Looking at it now

wtrocki commented:
```
 ! [remote rejected]       pr/RinkiyaKeDad/2248-1 -> pr/RinkiyaKeDad/2248-1 (permission denied)
error: failed to push some refs to 'git@github.com:RinkiyaKeDad/graphback'
```

I cannot fix issues in PR. Can you add me to fork :) THX

wtrocki commented:
> I tried verifying this by enabling the fastify template and then using the updated cli (by running

How about running this from the code:
```
cd templates/ts-fastify-mongodb-backend
docker-compose up -d
yarn develop
```

PS: This doesn't seem to be working for me.

RinkiyaKeDad commented:
> I think `resolvers` should include `noteResolvers` as in

Thanks for pointing that out 😅 I've added them now @machi1990.




> Can you add me to fork :) THX

Added you @wtrocki :)







> PS: This doesn't seem to be working for me.




Same. It doesn't give an error in the terminal like last time, but shows that localhost refused to connect on the browser.



![NewTest](https://user-images.githubusercontent.com/56963264/103456802-8d9af600-4d1f-11eb-9885-379e779afc1c.png)

RinkiyaKeDad commented:
> How about running this from the code:

Okay so this way of testing the template works if I don't specify the noteResolvers and simply leave the code like this:
```
  app.register((mercurius as any), {
    schema,
    resolvers,
    graphiql: true,
    context: contextCreator,
    subscription: true
  });
```
But even with this when I tested the subscriptions they were not working. Could it be because [`makeExecutableSchema`](https://github.com/aerogear/graphback/blob/master/templates/ts-fastify-mongodb-backend/src/index.ts#L39) is not creating the schema in a way acceptable by mercurius?

Will have to look for some other way to fix this then I guess 🤔

RinkiyaKeDad commented:
Would it be a good idea to switch from mercurius to [apollo-server-fastify](https://github.com/apollographql/apollo-server/tree/main/packages/apollo-server-fastify)?

wtrocki commented:
Sounds good?

RinkiyaKeDad commented:
Closing this since we're shifting to apollo-server-fastify. See #2256
